solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -210,6 +213,38 @@ public Replica getReplica(String coreNodeName) {
     return null;
   }
 
+  /**
+   * Check that all replicas in a collection are live
+   *
+   * @see CollectionStatePredicate
+   */
+  public static boolean isFullyActive(Set<String> liveNodes, DocCollection collectionState) {
+    Objects.requireNonNull(liveNodes);
+    if (collectionState == null)
+      return false;
+    for (Slice slice : collectionState) {
+      for (Replica replica : slice) {
+        if (replica.isActive(liveNodes) == false)
+          return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if the passed in DocCollection is null
+   *
+   * @see CollectionStatePredicate
+   */
+  public static boolean isDeleted(Set<String> liveNodes, DocCollection collectionState) {
+    return collectionState == null;
+  }
```

Comment:
maybe `exists`? isDeleted implies that it used to exist, but it may have never been created